crates/runtime/src/http/v1/chat.rs

```diff
@@ -41,6 +41,12 @@ pub(crate) async fn post(
     let span = tracing::span!(target: "task_history", tracing::Level::INFO, "ai_chat", input = %serde_json::to_string(&req).unwrap_or_default());
     span.in_scope(|| tracing::info!(target: "task_history", model = %req.model, "labels"));
 
+    if let Some(ref metadata) = req.metadata {
+        if let Some(serde_json::Value::String(trace_id)) = metadata.get("trace_id") {
```

Comment:
Shouldn't this come from a `traceparent` header instead? https://www.w3.org/TR/trace-context/#traceparent-header

Reply:
yeah that's a good point

Reply:
I have added this as a `traceparent` header. This means i've also added parent span id propagation.